### PR TITLE
chore(medusa, utils): rename buildLegacyFieldsListFrom to objectToStringPath

### PR DESCRIPTION
**Pull request by riqwan**
RESOLVES CORE-1317

### CODE REVIEW

**Review by adrien2p — APPROVED** [review]
LGTM, should we rename the other to stringPathToObject or keep it as it?

File: packages/medusa/src/repositories/customer-group.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ import {
   queryEntityWithIds,
   queryEntityWithoutRelations,
 } from "../utils/repository"
-import { buildLegacyFieldsListFrom } from "../utils"
+import { objectToStringPath } from "../utils"
```
Comment (olivermrbl) [inline packages/medusa/src/repositories/customer-group.ts:16]: todo: Import from `@medusajs/utils`

File: packages/medusa/src/repositories/order.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { flatten, groupBy, map, merge } from "lodash"
 import { FindManyOptions, FindOptionsRelations, In } from "typeorm"
 import { Order } from "../models"
-import { buildLegacyFieldsListFrom } from "../utils"
+import { objectToStringPath } from "../utils"
```
Comment (olivermrbl) [inline packages/medusa/src/repositories/order.ts:4]: todo: Import from `@medusajs/utils`

File: packages/medusa/src/repositories/product-category.ts [hunk 3]
```diff
@@ -9,7 +9,7 @@ import {
 import { ProductCategory } from "../models/product-category"
 import { ExtendedFindConfig, QuerySelector } from "../types/common"
 import { dataSource } from "../loaders/database"
-import { buildLegacyFieldsListFrom } from "../utils"
+import { objectToStringPath } from "../utils"
```
Comment (olivermrbl) [inline packages/medusa/src/repositories/product-category.ts:12]: todo: Import from `@medusajs/utils`

File: packages/medusa/src/repositories/product.ts [hunk 4]
```diff
@@ -10,7 +10,7 @@ import { ExtendedFindConfig } from "../types/common"
 import { dataSource } from "../loaders/database"
 import { ProductFilterOptions } from "../types/product"
 import {
-  buildLegacyFieldsListFrom,
+  objectToStringPath,
```
Comment (olivermrbl) [inline packages/medusa/src/repositories/product.ts:13]: todo: Import from `@medusajs/utils`

File: packages/medusa/src/repositories/tax-rate.ts [hunk 5]
```diff
@@ -16,7 +16,7 @@ import {
 } from "../models"
 import { TaxRateListByConfig } from "../types/tax-rate"
 import { isDefined } from "medusa-core-utils"
-import { buildLegacyFieldsListFrom } from "../utils"
+import { objectToStringPath } from "../utils"
```
Comment (olivermrbl) [inline packages/medusa/src/repositories/tax-rate.ts:19]: todo: Import from `@medusajs/utils`

File: packages/medusa/src/utils/build-query.ts [hunk 6]
```diff
@@ -153,7 +153,7 @@ function buildWhere<TWhereKeys extends object, TEntity>(
  * output: ['test.test1', 'test.test2', 'test.test3.test4', 'test2']
  * @param input
  */
-export function buildLegacyFieldsListFrom<TEntity>(
+export function objectToStringPath<TEntity>(
```
Comment (olivermrbl) [inline packages/medusa/src/utils/build-query.ts:156]: suggestion: Keep this and deprecate it to stay backward compatible. And change all occurrences to import from `@medusajs/utils`

**Review by olivermrbl — CHANGES_REQUESTED** [review]

**Review by olivermrbl — APPROVED** [review]
LGTM!

**Pull request merged**